tunnel/tunnel_normal.go

```diff
@@ -202,6 +219,36 @@ func (t NormalTunnel) generateAuthMethod(ctx context.Context) ([]ssh.AuthMethod,
 	return authMethods, nil
 }
 
+const keepaliveInterval = 1 * time.Minute
+const keepaliveTimeout = 15 * time.Second
+func sshKeepalive(ctx context.Context, client *ssh.Client, conn net.Conn, errChan chan <- error) {
+	t := time.NewTicker(keepaliveInterval)
+	defer t.Stop()
+
+	err := func() error {
+		for {
+			deadline := time.Now().Add(keepaliveInterval).Add(keepaliveTimeout)
```

Comment:
Should you set the deadline within `<-t.C` and not bake the keepaliveInterval into the deadline? Seems a bit more intuitive to me